app/views/assessment/AssessmentOption.js

```diff
@@ -0,0 +1,181 @@
+import DateTimePicker from '@react-native-community/datetimepicker';
+import React, { Fragment, useEffect, useMemo, useState } from 'react';
+import { StyleSheet, Text, TouchableOpacity, View } from 'react-native';
+import { SvgXml } from 'react-native-svg';
+
+import Icon from '../../assets/svgs/check';
+import Colors from '../../constants/colors';
+import Fonts from '../../constants/fonts';
+import { isPlatformAndroid, isPlatformiOS } from '../../Util';
+import {
+  SCREEN_TYPE_CHECKBOX,
+  SCREEN_TYPE_DATE,
+  SCREEN_TYPE_RADIO,
+} from './constants';
+
+/**
+ * @typedef { import("./Assessment").SurveyAnswers } SurveyAnswers
+ * @typedef { import("./Assessment").SurveyOption["values"][0] } SurveyOptionValue
+ */
+
+/** @type {React.FunctionComponent<{
+ *   answer?: SurveyAnswers[0][0];
+ *   index: number;
+ *   onSelect: (value: string) => void;
+ *   option: SurveyOptionValue;
+ *   selected: boolean;
+ *   type: Extract<SurveyScreen, 'Checkbox' | 'Date' | 'Radio'>
+ * }>} */
+const AssessmentOption = ({
+  answer,
+  index,
+  onSelect,
+  option,
+  selected,
+  type,
+}) => {
+  // The API doesn't have a defined way to know a specific option is a date,
+  // we just assume the first option is the date picker when type is SCREEN_TYPE_DATE
+  const isDate = type === SCREEN_TYPE_DATE && index === 0;
+  const showIndicator =
+    type === SCREEN_TYPE_CHECKBOX ||
+    type === SCREEN_TYPE_RADIO ||
+    type === SCREEN_TYPE_DATE;
+  const [showDatePicker, setShowDatePicker] = useState(false);
+  const [date, setDate] = useState(() =>
+    (isDate && selected && answer) ? new Date(answer.value) : null,
+  );
+  const label = useMemo(() => {
+    if (isDate && date)
```

Comment:
please use brackets to and a standard `if/else` to keep this more in line with our style guide.